broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionAdminAccess.java

```diff
@@ -140,4 +149,91 @@ public ActorFuture<Void> resumeProcessing() {
         });
     return completed;
   }
+
+  @Override
+  public ActorFuture<Void> banInstance(final long processInstanceKey) {
+    final ActorFuture<Void> future = concurrencyControl.createFuture();
+    concurrencyControl.run(
+        () -> {
+          try {
+            adminControl
+                .getLogStream()
+                .newLogStreamRecordWriter()
+                .onComplete(
+                    (writer, error) -> {
+                      if (error != null) {
+                        LOG.error(
+                            "Could not retrieve writer to write error record for process instance.",
+                            error);
+                        future.completeExceptionally(error);
+                        return;
+                      }
+
+                      writeErrorEventAndBanInstance(processInstanceKey, writer, future);
+                    });
+          } catch (final Exception e) {
+            LOG.error("Could not resume processing", e);
```

Comment:
:wrench: I missed this for the PR on `main` but this error message is not correct. Could you adjust it here and on main?